### ISSUE TITLE
Smart-match against Set object is not documented

### ISSUE THREAD

**Issue by titsuki** [perl6/doc#1182]
In the `Right-hand side` column of the following table:
https://docs.perl6.org/language/operators#index-entry-smart-match_operator
`Set:D` may be classified as `Any:D`.
However, in fact, `comparison semantics` of this operation is not `object identity`.
```
$ perl6 -e 'use Test; my Set \a = Set(1,2,3); my Set \b = Set(1,2,3); ok a ~~ b;'
ok 1 -  # comparison semantics is not object identity
$ perl6 -e 'use Test; my Hash \a = { a => 1 }; my Hash \b = { a => 1 }; ok a ~~ b';
not ok 1 -  # comparison semantics is object identity
```

See also:
+ https://github.com/rakudo/rakudo/blob/nom/src/core/Set.pm#L1
+ https://github.com/rakudo/rakudo/blob/nom/src/core/Setty.pm#L70-L72

**Comment by titsuki** [comment]
These are tests for this operation:
https://github.com/perl6/roast/blob/master/S02-types/set.t#L44-L56

**Comment by zoffixznet** [comment]
No idea about docs, but it's possible your Hash comparison above is buggy: https://irclog.perlgeek.de/perl6-dev/2017-02-12#i_14087552

Will look into it deeper in the next 48 hours.

**Comment by gfldex** [comment]
fixed with 27c280a81f
